pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputer.java

```diff
@@ -65,6 +70,26 @@ public int computeThreshold(StreamConfig streamConfig, CommittingSegmentDescript
     long optimalSegmentSizeBytesMin = desiredSegmentSizeBytes / 2;
     double optimalSegmentSizeBytesMax = desiredSegmentSizeBytes * 1.5;
 
+    double segmentSizeVariation = DEFAULT_FLUSH_THRESHOLD_VARIANCE_PERCENTAGE;
+    try {
+      if (streamConfig.getStreamConfigsMap().containsKey(FLUSH_THRESHOLD_VARIANCE_PERCENTAGE)) {
+        segmentSizeVariation =
+            Double.parseDouble(streamConfig.getStreamConfigsMap().get(FLUSH_THRESHOLD_VARIANCE_PERCENTAGE));
+      }
+    } catch (Exception e) {
+      SegmentSizeBasedFlushThresholdUpdater.LOGGER.warn(
+          "Invalid value for segment size variation: {}. Using default {}",
+          streamConfig.getStreamConfigsMap().get(FLUSH_THRESHOLD_VARIANCE_PERCENTAGE),
+          DEFAULT_FLUSH_THRESHOLD_VARIANCE_PERCENTAGE);
+    }
+    if (segmentSizeVariation < 0.0 || segmentSizeVariation >= 1.0) {
```

Comment:
This is not percentage. Percentage should be between 0-100

Reply:
fraction